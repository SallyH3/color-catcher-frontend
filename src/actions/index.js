export const makePalette = (palette) => {
	return {
		type: 'MAKE_PALETTE',
		palette
	}
}

export const clearPalette = (palette) => {
	return {
		type: 'CLEAR_PALETTE',
		palette
	}
}