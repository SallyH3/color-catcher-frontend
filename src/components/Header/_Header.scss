header {
	display: flex;
	justify-content: space-between;
	background-color: blue;
}

.generate {
	height: 20px;
	margin-top: 10px;
	
}