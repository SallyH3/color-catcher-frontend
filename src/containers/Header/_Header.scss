header {
	display: flex;
	justify-content: space-between;
	background-color: #EF6A6A;
}

.generate {
	height: 20px;
	margin-top: 10px;
	
}