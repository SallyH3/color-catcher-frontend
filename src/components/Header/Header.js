import React from 'react';
import './_Header.scss';

export const Header = () => {
	return(
		<header>
			<h1>Color Catcher</h1>
			<button className='generate'>Generate Colors</button>
		</header>
	)
}