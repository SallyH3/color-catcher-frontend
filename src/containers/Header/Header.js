import React, { Component } from 'react';
import './_Header.scss';
import { makePalette, clearPalette } from '../../actions';
import { connect } from 'react-redux';

class Header extends Component {
	constructor() {
		super();
		this.state = {
			colors: []
		}
	}

	componentDidMount = () => {
		this.generateColors()
	}
	
	handleClick = () => {
		this.setState({colors: []})
		this.generateColors()
	}

	generateColors = () => {
		let colorObject = {locked: false}
		for(let i = 0; i < 5; i++) {
		const characters = "0123456789ABCDEF";
		let color = '#'
			for(let i = 0; i< 6; i++)
				color+= characters[(Math.floor(Math.random() * 16))]
			colorObject = {...colorObject, color}
			this.state.colors.push(colorObject)
		}
		this.props.makePalette(this.state.colors);
	}

	render() {
		return(
			<main>
				<header>
					<h1>Color Catcher</h1>
					<button className='generate' onClick={this.handleClick} >Generate Colors</button>
				</header>
			</main>
		)
	}
}

const mapStateToProps = (state) => ({
	palette: state.palette
});

const mapDispatchToProps = (dispatch) => ({
	makePalette: (palette) => dispatch(makePalette(palette)),
 	clearPalette: (palette) => dispatch(clearPalette(palette))
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);