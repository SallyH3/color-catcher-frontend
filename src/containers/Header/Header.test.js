import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapStateToProps, mapDispatchToProps } from './Header';
import { makePalette } from '../../actions';

describe('Header', () => {
	let wrapper;
	let mockMakePalette;
	let mockPalette;

	beforeEach(() => {
		mockMakePalette = jest.fn();
		mockPalette = [
			{	locked: false,
				color: "#909C0E"
			},
			{	locked: false,
				color: "#82BBC8"
			},
			{	locked: false,
				color: "#82BBC8"
			},
			{
				locked: false,
				color: "#82BBC8"
			},
			{	locked: false,
				color: "#909C0E"
			}
		]

		wrapper = shallow(
			<Header
				makePalette={mockMakePalette}
				palette={mockPalette}
			/>
		);
	});

	it('should match snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	describe('mapStateToProps', () => {
		it('should return an array of objects', () => {
			const mockState = {
					palette: [
				{	locked: false,
					color: "#909C0E"
				},
				{	locked: false,
					color: "#82BBC8"
				},
				{	locked: false,
					color: "#82BBC8"
				},
				{
					locked: false,
					color: "#82BBC8"
				},
				{	locked: false,
					color: "#909C0E"
				}
			],
				projects: []
		}

			const expected = {
				palette: [
			{	locked: false,
				color: "#909C0E"
			},
			{	locked: false,
				color: "#82BBC8"
			},
			{	locked: false,
				color: "#82BBC8"
			},
			{
				locked: false,
				color: "#82BBC8"
			},
			{	locked: false,
				color: "#909C0E"
			}
		]
	}

		const mappedProps = mapStateToProps(mockState);

		expect(mappedProps).toEqual(expected);
		});
	});

	describe('mapDispatchToProps', () => {
		it('should call dispatch when using makePalette', () => {
			const mockPalette = [
				{	locked: false,
					color: "#909C0E"
				},
				{	locked: false,
					color: "#82BBC8"
				},
				{	locked: false,
					color: "#82BBC8"
				},
				{
					locked: false,
					color: "#82BBC8"
				},
				{	locked: false,
					color: "#909C0E"
				}
			]
		
		const mockDispatch = jest.fn();

		const actionToDispatch = makePalette(mockPalette);
		const mappedProps = mapDispatchToProps(mockDispatch);

		mappedProps.makePalette(mockPalette)

		expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
		});
	});
});