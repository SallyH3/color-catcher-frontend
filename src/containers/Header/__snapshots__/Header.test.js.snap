// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<main>
  <header
    className="header-container"
  >
    <h1
      className="header-title"
    >
      Color Catcher
    </h1>
    <button
      className="generate"
      onClick={[Function]}
    >
      Generate Colors
    </button>
  </header>
</main>
`;
